packages/react-avatar/SPEC.md

```diff
@@ -0,0 +1,305 @@
+```
+
+## Variants
+
+### Color
+
+The Avatar supports color variants when displaying initials or an icon:
+
+- **Neutral** - Gray (default)
+- **Brand** - Brand colors from the theme
+- **Colorful** - Pick from a list of pre-defined Avatar colors. The color will be assigned based on a hash of the name
+  (to "randomly" assign a person a color). The color index can also be specified explicitly in case the use case
+  requires a different algorithm to pick the color.
+
+### Shape
+
+The Avatar natively supports a circular and square shape. However, some use cases require a hexagon shape (e.g. to represent a bot). The hexagon will not be supported directly, but it is possible to style the Avatar with a hexagon background by using `tokens`:
+
+```jsx
+<Avatar
+  icon={<ChatBotIcon />}
+  display="icon"
+  tokens={{
+    width: 'calc(var(--avatar-height) * 1.125)',
+    background: `url('${hexagonSvg}') 0px/contain no-repeat`,
+    borderRadius: '0',
+  }}
+/>
+```
+
+Where hexagonSvg is defined as:
+
+```js
+hexagonSvg =
+  'data:image/svg+xml;utf8,' +
+  '<svg width="36" height="32" viewBox="0 0 36 32" fill="none" xmlns="http://www.w3.org/2000/svg">' +
+  '<path fill="rgb(232,232,232)" d="M0.407926 17.528C-0.135976 16.5859 -0.135975 15.4141 0.407926 14.472' +
+  'L7.91541 1.46793C8.44076 0.557947 9.39444 0 10.4245 0H25.5755C26.6056 0 27.5592 0.557951 28.0846 1.46793' +
+  'L35.5921 14.472C36.136 15.4141 36.136 16.5859 35.5921 17.528L28.0846 30.5321' +
+  'C27.5592 31.4421 26.6056 32 25.5755 32H10.4245C9.39443 32 8.44076 31.4421 7.91541 30.5321L0.407926 17.528Z"/>' +
+  '</svg>';
+```
+
+## API
+
+From [Avatar.types.tsx](https://github.com/microsoft/fluentui/blob/master/packages/react-avatar/src/components/Avatar/Avatar.types.tsx):
+
+### Slots
+
+- `as` - The root element of the Avatar.
+- `image` - The Avatar's image, if available.
+- `label` - The text shown when there's no image. Defaults to the initials derived from `name` using `getInitials`.
+- `icon` - Icon displayed when there's no image or intials available.
+- `badge` - Badge to show the avatar's status.
+
+### Props
+
+```ts
+export interface AvatarProps extends React.HTMLAttributes<HTMLElement> {
+    /**
+   * The root element type of the Avatar.
+   *
+   * @defaultvalue span
+   */
+  as?: React.ElementType;
+
+  /** The Avatar's image. */
+  image?: ShorthandValue<{}>;
+
+  /** The label shown when there's no image or icon. Defaults to the initials derived from `name` using `getInitials`. */
+  label?: ShorthandValue<{}>;
+
+  /** Icon displayed when there's no image. */
+  icon?: ShorthandValue<{}>;
+
+  /** Badge to show the avatar's status. */
+  badge?: ShorthandValue<BadgeProps>;
+
+  /** The name used for displaying the initials of the avatar if the image is not provided. */
+  name?: string;
+
+  /** Custom method for generating the initials from the name property, which is shown if no image is provided. */
+  getInitials?: (name: string, isRtl: boolean) => string;
```

Comment:
Will this function be the current implementation for persona (https://github.com/microsoft/fluentui/blob/master/packages/utilities/src/initials.ts) or do you plan to have a different function here?

Reply:
Yes, Avatar will use the function you linked as the default implementation for `getInitials`. This prop allows you to optionally override and provide a different implementation if needed.